pkg/agent/handler/credential.go

```diff
@@ -123,37 +136,119 @@ func (h *credentials) getManagedApp(cred *mv1.Credential) (*v1.ResourceInstance,
 	return h.client.GetResource(url)
 }
 
-type creds struct {
+func (h *credentials) getCRD(cred *mv1.Credential) (*mv1.CredentialRequestDefinition, error) {
+	url := fmt.Sprintf(
+		"/management/v1alpha1/environments/%s/managedapplications/%s",
+		cred.Metadata.Scope.Name,
+		cred.Spec.CredentialRequestDefinition,
+	)
+	ri, err := h.client.GetResource(url)
+	if err != nil {
+		return nil, err
+	}
+
+	crd := &mv1.CredentialRequestDefinition{}
+	err = crd.FromInstance(ri)
+	return crd, err
+}
+
+// encryptMap loops through all provided data and checks the value against the provisioning schema to see if it should be encrypted.
+func encryptMap(sec cat.ApplicationSpecSecurity, schema, data map[string]interface{}) map[string]interface{} {
+	for key, value := range data {
+		schemaValue := schema[key]
+		v, ok := schemaValue.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		if _, ok := v[xAgentEncrypted]; ok {
+			v, ok := value.(string)
+			if !ok {
```

Comment:
Should we check the value as well not just the existence of x-agent-encrypted property?